### PR TITLE
refactor: expose generated code and client business logic to share with Kapacitor

### PR DESCRIPTION
**Pull request by lesam**

### CODE REVIEW

**Comment by lesam** [comment]
Depends on: https://github.com/influxdata/influx-cli/pull/102

**Comment by danxmoran** [comment]
Holding off on a full review until #102 merges and this rebases, to make it easier to see the truly new stuff.

Can you summarize all the moves? I see:
* `internal/api` -> `api`
* `internal/cmd` -> `clients`

Any others?

File: clients/bucket/delete_test.go
Comment (danxmoran) [inline clients/bucket/delete_test.go:6]: Bad import sort (I'd expect `make fmt` to fix it, let me know if not)

File: internal/mock/gen.go
Comment (danxmoran) [inline internal/mock/gen.go:15]: Should the config service also move out of `internal`?

Comment (lesam) [reply internal/mock/gen.go:15]: I don't need it for now, I vote we leave it in `internal`

File: clients/bucket/delete_test.go
Comment (lesam) [reply clients/bucket/delete_test.go:6]: Nope, `make fmt` was already run on this.

**Review by danxmoran — APPROVED** [review]

**Pull request merged**
